pkg/asset/quota/aws/aws.go

```diff
@@ -137,7 +138,12 @@ func machineTypeToQuota(t string, instanceTypes map[string]InstanceTypeInfo) quo
 	switch class {
 	case "a", "c", "d", "h", "i", "m", "r", "t", "z":
 		return quota.Constraint{Name: "ec2/L-1216C47A", Count: info.vCPU}
+	case "g":
+		return quota.Constraint{Name: "ec2/L-3819A6DF", Count: info.vCPU}
 	default:
+		if strings.HasPrefix(strings.ToLower(t), "vt") {
+			return quota.Constraint{Name: "ec2/L-3819A6DF", Count: info.vCPU}
+		}
```

Comment:
What is unique about the VT instance class that we are looking for prefixes rather than the entire string?
I would have expected the following.
```suggestion
	case "vt":
		return quota.Constraint{Name: "ec2/L-3819A6DF", Count: info.vCPU}
	default:
```

Reply:
Class was set to t[0] so I thought just the first character is being set to it

Reply:
Ah, right. That makes the code awkward. I think I can live with what you have, but give me a bit to ruminate on it.

Reply:
The logic of just looking at the first character is flawed to begin with, unfortunately. For example, the DL instance types are incorrectly interpreted as part of the D class instead of the DL class. I was hoping that we could use a regex with the assumption that the instance class is always followed by a number (eg, M4), but there are the Im4gn and Is4gen types that are part of the I instance class. So then I thought maybe a number or a lower-case letter, but there are the Inf1 types that are part of the Inf instance class.

Nevertheless, I propose that we are better off with something akin to the following.
https://go.dev/play/p/OBx-EcfMjVS